node/cn/backend.go

```diff
@@ -313,18 +313,24 @@ func New(ctx *node.ServiceContext, config *Config) (*CN, error) {
 		reward.NewStakingManager(cn.blockchain, governance, cn.chainDB)
 	}
 
-	var restartFn func()
-	if config.AutoRestartFlag {
-		restartFn = func() {
-			daemonPath := config.DaemonPathFlag
-			logger.Warn("Restart node", "command", daemonPath+" restart")
-			cmd := exec.Command(daemonPath, "restart")
-			cmd.Run()
+	// set worker
+	if config.WorkerDisable {
+		cn.miner = work.NewFakeWorker()
+	} else {
+		var restartFn func()
+		if config.AutoRestartFlag {
+			restartFn = func() {
+				daemonPath := config.DaemonPathFlag
+				logger.Warn("Restart node", "command", daemonPath+" restart")
+				cmd := exec.Command(daemonPath, "restart")
+				cmd.Run()
+			}
 		}
```

Comment:
Is this logic need to be inside of `else` branch?

Reply:
The restart logic is run in worker. If worker is disabled, the restart is not run.
I am planning on moving the restart routine outside the miner.

Reply:
Okay. FakeWorker disables not only restart logic but other logics also though.